src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -325,6 +325,28 @@ internal virtual NamedTypeSymbol LookupMetadataType(ref MetadataTypeName emitted
                 }
             }
 
+            if (isTopLevel && MetadataHelpers.DecodeFileType(emittedTypeName.UnmangledTypeName) is (not -1 and var ordinal, var typeName))
+            {
+                // also do a lookup for file types from source.
+                namespaceOrTypeMembers = scope.GetTypeMembers(typeName);
+                foreach (var named in namespaceOrTypeMembers)
+                {
+                    if (named is SourceMemberContainerTypeSymbol { AssociatedSyntaxTree: SyntaxTree tree }
```

Comment:
>named is SourceMemberContainerTypeSymbol [](http://example.com/codeflow?start=24&length=40)

It looks like this code is not going to work for retargeting scenario

Reply:
Removed the check for `SourceMemberContainerTypeSymbol` and added a test plan (#60819) item *Test getting a retargeted file type from source using its metadata name*.